File: src-tauri/service/src/models/invoice.rs
use crate::{NewInvoiceItem, UpdateInvoiceItem};
use sea_orm::FromQueryResult;
use serde::{Deserialize, Serialize};

#[derive(Deserialize, Serialize, Debug, PartialEq, FromQueryResult)]
pub struct SelectInvoices {
    pub id: String,
    pub created_at: String,
    pub paid_amount: f64,
    pub client_id: String,
    pub full_name: String,
    pub status: String,
    pub identifier: String,
    pub products: i64,
    pub total: f64,
}

#[derive(Deserialize, Serialize, Debug, PartialEq, FromQueryResult)]
pub struct SelectInvoiceDetails {
    pub id: String,
    pub order_id: String,
    pub created_at: String,
    pub paid_amount: f64,
    pub full_name: String,
    pub address: Option<String>,
    pub phone_number: Option<String>,
    pub email: Option<String>,
    pub status: String,
    pub identifier: String,
    pub total: f64,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct NewInvoice {
    pub client_id: String,
    pub order_id: Option<String>,
    pub status: String,
    pub paid_amount: f64,
    pub items: Vec<NewInvoiceItem>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct UpdateInvoice {
    pub id: String,
    pub client_id: String,
    pub status: String,
    pub paid_amount: f64,
    pub items: Vec<UpdateInvoiceItem>,
}

File: types/invoice.ts
import type { INVOICE_STATUSES } from "~/consts/status";

export interface InvoiceT {
  id: string;
  status: (typeof INVOICE_STATUSES)[number];
  paid_amount: number;
  client_id: string;
  order_id: string;
  full_name?: string;
  products?: number;
  total?: number;
  created_at?: string;
  identifier: string;
}

export interface InvoiceForUpdateT {
  id: string;
  client_id: string;
  full_name: string;
  paid_amount: number;
  created_at: string;
  status: (typeof INVOICE_STATUSES)[number];
  items: {
    id: string;
    inventory_id: string;
    name: string;
    product_id: string;
    quantity: number;
    price: number;
  }[];
}

export interface InvoiceProductsPreviewT {
  price: number;
  name: string;
  quantity: number;
}
